packages/tpetra/tsqr/cmake/Dependencies.cmake

```diff
@@ -6,3 +6,10 @@ SET(LIB_REQUIRED_DEP_TPLS)
 SET(LIB_OPTIONAL_DEP_TPLS CUBLAS CUSOLVER)
 SET(TEST_REQUIRED_DEP_TPLS)
 SET(TEST_OPTIONAL_DEP_TPLS)
+
+IF(DEFINED TPL_ENABLE_CUDA)
```

Comment:
You don't need this IF() statement, do you?  Don't you just need `IF (TPL_ENABLE_CUDA)`?  Is there some behavior of CMake you are seeing that I am not aware of?

Reply:
@bartlettroscoe I'm just being excessively cautious, but if you like, I can remove the outer `IF`.

Reply:
@mhoemmen, for better or for worse, CMake evaluates undefined variables as empty "" which evaluates to `FALSE` so:

```
IF (DEFINED <SOME_VAR>)
  IF (<SOME_VAR>)
     ...
  ENDIF()
ENDIF()
```

and

```
IF (<SOME_VAR>)
     ...
ENDIF()
```

should be identical

Reply:
@bartlettroscoe This was a problem historically when we did things like this:
```
IF(FOO AND BAR)
```
where FOO was not defined but BAR was.  I'll change this, though.